test/uk/gov/hmrc/helptosavefrontend/controllers/RegisterControllerSpec.scala

```diff
@@ -94,6 +96,45 @@ class RegisterControllerSpec extends TestSupport with EnrolmentAndEligibilityChe
         contentAsString(result) should include(validNSIUserInfo.surname)
       }
 
+      "show the users details with the verified user email address " +
+        "if the user has not already enrolled and " +
+        "the session data shows that they have been already found to be eligible " +
+        "and the user has clicked on the verify email link sent to them by the email verification service " in {
+        val testEmail = "email@gmail.com"
+        val theNino = "AE1234XXX"
+        inSequence{
+          mockPlayAuthWithRetrievals(AuthWithConfidence)(userDetailsURIWithEnrolments)
+          mockEnrolmentCheck(nino)(Right(EnrolmentStatus.NotEnrolled))
+          mockSessionCacheConnectorGet(Right(Some(HTSSession(Some(validNSIUserInfo.copy (nino = theNino)), None))))
+        }
+        val params = EmailVerificationParams(theNino, testEmail)
+        val result = doRequestWithQueryParam(params.encode().replaceAll("%2b", "+"))
```

Comment:
would a URL-decode here work to catch all URL-encoded characters? (there's `java.net.URLDecoder` you could use)